modules/pubsub/variables.tf

```diff
@@ -76,3 +76,9 @@ variable "kms_key_name" {
   type        = string
   default     = null
 }
+
+variable "message_storage_policy" {
+  type        = map(any)
```

Comment:
@essentialmetrics can you plz define this as an object with optional components?